app/api/scraper.py

```diff
@@ -2,23 +2,23 @@
 
 from app import schemas
 from app.services.scrape import TwitterScraper
+from app.services.url_parser import parse_twitter_url
 
 router = APIRouter()
 
 
 @router.get("/check", response_model=schemas.CheckResponse, name="user:get-data")
 def user_info_check(url: str, scraper: TwitterScraper = Depends()):
-    # Validate input url
-    if url[:20] == "https://twitter.com/":
-        username = url.split("/")[3]
-    elif url[:12] == "twitter.com/":
-        username = url.split("/")[1]
-    elif url != "" and "/" not in url:
-        username = url
-    else:
+    try:
+        twitter_entity, twitter_id = parse_twitter_url(url)
+    except ValueError:
         raise HTTPException(status_code=400, detail="'url' argument is invalid!")
 
-    user_db = scraper.get_user_by_username(username)
+    if twitter_entity == "tweet":
+        tweet = scraper.api_v2.get_tweet(twitter_id, tweet_fields=["author_id"])
+        user_db = scraper.get_user_by_id(tweet.data.author_id)
+    else:
+        user_db = scraper.get_user_by_username(twitter_id)
```

Comment:
Just realize that in the `tweetUrl` contains the username :D, so we can just get the username and everything is back to normal ✌️. No need to the extra flow `twitter_entity == 'tweet`

Reply:
Why not need extra flow?

Reply:
Cause if we just get the `username` from the `twitterUrl`, things get simple?

Reply:
There is no `twitterUrl` in this function. Besides, we need to support tweet URL too.

Reply:
yes by `twitterUrl` i ment tweet URL (sr for confusion). Point being we can get the account username from the tweet URL. It simplifies everything that way!